packages/dashboard/src/app/router/routerProvider.js

```diff
@@ -63,7 +65,7 @@ function RouterProvider({ children, ...props }) {
   };
 
   const [queryParams, setQueryParams] = useState(
-    parse(history.current.location.search)
+    parse(history.current?.location?.search)
```

Comment:
If `history.current?.location?.search` is `null` then this will be `parse(null)` which will return `{ null: '' }`, which seems unexpected and like something that could possible cause bugs.

Might be worth adding JS unit tests with

```js
/**
 * @jest-environment node
 */
```

to test this provider

Reply:
I am thinking, what would be the possible case when `history.current?.location?.search` may be `null`. We are only adding `.?` here to prevent errors during server-side rendering.

Reply:
During server-side rendering it will be `null`, so the initial state value would be `null` here, no?

Reply:
Ah indeed, if it's `undefined` and not `null` then it should be fine.

Reply:
Was trying to add test coverage for `@jest-environment node` however it doesn't look straightforward as it wouldn't run the same way as it does in SSR. When running a test in a node environment it will also run code in `useEffect`.

Though [React Hooks Testing Library](https://react-hooks-testing-library.com/) does provide [SSR support](https://react-hooks-testing-library.com/usage/ssr) for `renderHook`, however that doesn't seem to work as expected. Let me see if I can find a solution.